crates/runestick/src/interface.rs

```diff
@@ -34,8 +34,35 @@ impl Interface {
         Iterator::from_value(value)
     }
 
+    /// Retrieves the type name of the type
+    pub fn into_type_name(self) -> Result<String, VmError> {
+        let hash = Hash::instance_function(self.target.type_hash()?, Protocol::INTO_TYPE_NAME);
+        if let Some(name) = self.context.constant(hash) {
+            match name {
+                ConstValue::String(s) => return Ok(s.clone()),
+                ConstValue::StaticString(s) => return Ok((*s).to_string()),
+                _ => Err(VmError::expected::<String>(name.type_info()))?,
+            }
+        }
+
+        let hash = Hash::instance_function(self.target.type_hash()?, Protocol::INTO_TYPE_NAME);
+        if let Some(name) = self.unit.constant(hash) {
+            match name {
+                ConstValue::String(s) => return Ok(s.clone()),
+                ConstValue::StaticString(s) => return Ok((*s).to_string()),
+                _ => Err(VmError::expected::<String>(name.type_info()))?,
+            }
```

Comment:
This can be replaced with:
```rust
return String::from_value(name);
```
See: https://github.com/rune-rs/rune/blob/main/crates/runestick/src/from_value.rs#L147

Reply:
Unfortunately not, but maybe it should... This is ConstValue, not Value.

Reply:
Aha. Right :)